signing_clients/apps.py

```diff
@@ -79,22 +77,43 @@ def ignore_certain_metainf_files(filename):
     return False
 
 
-def file_key(zinfo):
-    '''
-    Sort keys for xpi files
-    @param name: name of the file to generate the sort key from
-    '''
-    # Copied from xpisign.py's api.py and tweaked
-    name = zinfo.filename
+def file_key(filename):
+    """Sort keys for xpi files
+
+    The filenames in a manifest are ordered so that files not in a
+    directory come before files in any directory, ordered
+    alphabetically but ignoring case, with a few exceptions
+    (install.rdf, chrome.manifest, icon.png and icon64.png come at the
+    beginning; licenses come at the end).
+
+    This order does not appear to affect anything in any way, but it
+    looks nicer.
+    """
     prio = 4
-    if name == 'install.rdf':
+    if filename == 'install.rdf':
         prio = 1
-    elif name in ["chrome.manifest", "icon.png", "icon64.png"]:
+    elif filename in ["chrome.manifest", "icon.png", "icon64.png"]:
         prio = 2
-    elif name in ["MPL", "GPL", "LGPL", "COPYING", "LICENSE", "license.txt"]:
+    elif filename in ["MPL", "GPL", "LGPL",
+                      "COPYING", "LICENSE", "license.txt"]:
```

Comment:
Let's make this list maybe a constant and pull it out to the top of the file so that it becomes more manageable.